crates/turbopack-ecmascript/src/chunk/mod.rs

```diff
@@ -625,7 +628,11 @@ impl EcmascriptChunkContentVc {
         }
         code += "]);\n";
         if this.evaluate.is_some() {
-            let runtime_code = embed_file!("js/src/runtime.js").await?;
+            let runtime_code = match *this.environment.chunk_loading().await? {
+                ChunkLoading::None => return Err(anyhow!("unsupported environment")),
+                ChunkLoading::Cjs => embed_file!("js/src/runtime.cjs.js").await?,
+                ChunkLoading::Dom => embed_file!("js/src/runtime.dom.js").await?,
```

Comment:
How about placing the common part into a common file and concatenate this with the unique parts.

Reply:
This is definitely possible. There are a few constructs that are different, e.g. HMR does not make sense in a server context (yet?) so I removed it, so `SourceType.Update` would not exist. However, it's not much of a concern right now.

Reply:
Let's just keep HMR in the server context too. We eventually want to add that anyway. There is currently no source that trigger that anyway.